### ISSUE TITLE
Seperate `dependencies` and `devDependencies`

### ISSUE THREAD

**Issue by tech4him1** [netlify/netlify-cms#589]
On dependencies, we need to decide whether to:
1. Include the sources+dist in the npm package, leave all dependencies how they are.
2. Move all dependencies to devDependencies, only have the bundle in the npm package.

or some combination of the above.

**Comment by tech4him1** [comment]
cc/@Benaiah @erquhart

**Comment by dopry** [comment]
As a consumer of netlify, not as a developer I'd want the smallest package footprint possible.  As a developer I'd want to include source maps to facilitate support. 

Given those criteria, I'd vote to move everything to devDependencies, and just distribute a bundle with source maps for debugging.

**Comment by Benaiah** [comment]
@dopry part of the issue is that distributing a bundle _increases_ the final package footprint if you import as an npm package due to library duplication (if you have any of the same deps as the CMS, they'll be duplicated in your final bundle since you're importing already-bundled code).

**Comment by dopry** [comment]
I see. I was thinking more as just person who wanted to include Netlify-CMS for my blog. I guess most users like that could just load from CDN. Most npm users would probably be developers looking for a library to include in other react projects, eh?  In that case I'd probably lean toward making sure only run time deps are in dependencies.  It does look like a devDependencies have snuck into deps, like node-sass and babel-plugin-transform-builtin-extend...

**Comment by tech4him1** [comment]
We should be able to do something like run `babel` without `webpack` for the NPM module.

**Comment by tech4him1** [comment]
Helpful read: http://2ality.com/2017/04/setting-up-multi-platform-packages.html

**Comment by tech4him1** [comment]
> We should be able to do something like run `babel` without `webpack` for the NPM module.

Currently, I'm only seeing a few "special" loaders we'd need to deal with before I can do this:
- `svg-loader`: not really anything special, just minifying and inlineing as a JS string
  - not hard
- `css-loader`: only used to import default styles for `redux-notifications` and `react-datetime`
  - These are *default* styles, not required. What do you guys think about just pulling the styles into our stylesheets at build time, but leaving the JS for those packages out (for the bundler to do)?

**Comment by erquhart** [comment]
Makes sense to me. I ran into issues that led me to stick with Webpack, but if I had had more time, I would have pushed for Rollup or Babel. Worth digging into.

**Comment by martinjagodic** [comment]
Closing as stale and/or outdated.
